src/poetry/installation/executor.py

```diff
@@ -158,11 +160,41 @@ def execute(self, operations: list[Operation]) -> int:
                     operation.package.develop
                     and operation.package.source_type in {"directory", "git"}
                 )
-                if not operation.skipped and is_parallel_unsafe:
+
+                if operation.skipped:
+                    # Skipped operations are safe to execute in parallel
+                    tasks.append(
+                        self._executor.submit(self._execute_operation, operation)
+                    )
+                elif is_parallel_unsafe:
                     serial_operations.append(operation)
-                    continue
+                elif operation.package.source_type == "git":
+                    # Git operations on the same repository should be executed serially
+                    serial_git_operations[operation.package.source_url].append(
+                        operation
+                    )
+                else:
+                    tasks.append(
+                        self._executor.submit(self._execute_operation, operation)
+                    )
 
-                tasks.append(self._executor.submit(self._execute_operation, operation))
+            # For each git repository, execute all operations serially
+            for repository_git_operations in serial_git_operations.values():
+
+                def _serialize(
+                    repository_serial_operations: list[Operation],
+                ) -> None:
+                    for operation in repository_serial_operations:
+                        self._execute_operation(operation)
+
+                tasks.append(
+                    self._executor.submit(
+                        functools.partial(
+                            _serialize,
+                            repository_serial_operations=repository_git_operations,
```

Comment:
Is `functools.partial` an unnecessary complication here? Isn't this code just equivalent to
```python
tasks.append(self.executor.submit(_serialize, repository_git_operations))
```
?

Reply:
You are right! Fixed it now.